spring-cloud-gcp-data-spanner/src/main/java/com/google/cloud/spring/data/spanner/repository/query/SqlSpannerQuery.java

```diff
@@ -232,15 +233,33 @@ private List executeReadSql(Pageable pageable, Sort sort, QueryTagValue queryTag
 
 		Statement statement = buildStatementFromQueryAndTags(queryTagValue);
 
-		return (getReturnedSimpleConvertableItemType() != null)
-				? this.spannerTemplate.query(
-						struct -> new StructAccessor(struct).getSingleValue(0), statement,
-						spannerQueryOptions)
-				: this.spannerTemplate.query(this.entityType,
-						statement,
+		if (getReturnedSimpleConvertableItemType() != null) {
+			return this.spannerTemplate.query(
+					struct -> new StructAccessor(struct).getSingleValue(0), statement,
+					spannerQueryOptions);
+		}
+		// check if is json field
+		String jsonFieldName = getJsonFieldNameIfExists(returnedType);
```

Comment:
Could you explain what this part of the code does? Looks like there is a way to query single column?

Reply:
The part of the code determines how the query result get processed. If returned type is for example String or List<String>, versus Entity or List<Entity>. Now with JSON fields, there is a new possibility that the returned type is a custom POJO class that's annotated with json. 

This is an edge case that cannot be handled by the original logic, the code checks if the `returnedType` is one of the json field/s in the entity, if so, gets the corresponding field name from `jsonPropertiesClassToName` and proceed with `getSingleJsonValue` structAccessor function. 

Here is an [integration test](https://github.com/GoogleCloudPlatform/spring-cloud-gcp/blob/41be5213ed7e5bee7695f247604a2f8ae61b35a9/spring-cloud-gcp-data-spanner/src/test/java/com/google/cloud/spring/data/spanner/repository/it/SpannerRepositoryIntegrationTests.java#L531-L533) added for this edge case.